MapboxCoreNavigationTests/NavigationServiceTests.swift

```diff
@@ -617,6 +617,110 @@ class NavigationServiceTests: XCTestCase {
+            "Haymarket",
+            "Charles II Street #1",
+            "Charles II Street #2",
+            "Regent Street St James\'s",
+            "Regent Street #1",
+            "Regent Street #2",
+            "Air Street",
+            "Brewer Street",
+            "Lexington Street",
+            "Broadwick Street #1",
+            "Broadwick Street #2",
+            "Dufour\'s Place",
+            "Dropped Pin #1"
+        ]
+        
+        if let route = route {
+            XCTAssertEqual(route.legs[0].steps.count, 11, "Unexpected number of steps.")
+            
+            let navigationService = MapboxNavigationService(route: route, routeOptions: routeOptions, simulating: .never)
+            let routeController = navigationService.router as! RouteController
+            
+            // Iterate over each step and simulate movement along the route.
+            route.legs[0].steps.enumerated().forEach {
+                let stepCoordinates = $0.element.shape!.coordinates
+                let stepLocations: [CLLocation] = stepCoordinates.enumerated().map {
+                    let currentCoordinate = $0.element
+                    return CLLocation(latitude: currentCoordinate.latitude, longitude: currentCoordinate.longitude)
+                }
+                
+                stepLocations.enumerated().forEach {
+                    let currentSpokenInstruction = navigationService.routeProgress.currentLegProgress.currentStepProgress.currentSpokenInstruction!.text
+                    XCTAssertFalse(voiceInstructionsArray.isEmpty, "Voice instructions array should not be empty.")
+                    if currentSpokenInstruction != voiceInstructionsArray.first {
+                        voiceInstructionsArray.removeFirst()
+                    }
+
+                    XCTAssertEqual(currentSpokenInstruction, voiceInstructionsArray.first)
+                    
+                    let currentBannerInstruction = navigationService.routeProgress.currentLegProgress.currentStepProgress.currentVisualInstruction!.primaryInstruction.text!
+                    XCTAssertFalse(bannerInstructionsArray.isEmpty, "Banner instructions array should not be empty.")
+                    if currentBannerInstruction != bannerInstructionsArray.first {
+                        bannerInstructionsArray.removeFirst()
+                    }
+                    
+                    XCTAssertEqual(currentBannerInstruction, bannerInstructionsArray.first)
+                    
+                    // Attempt to snarf instructions by calling methods which internally call Navigator.getStatusForTimestamp(_:) multiple times.
+                    let outOfBandCallsCount = 5
+                    for _ in 0..<outOfBandCallsCount {
+                        let _ = navigationService.router.userIsOnRoute($0.element)
+                        let _ = routeController.location
+                        let _ = routeController.snappedLocation
```

Comment:
How do you think, maybe it is better to introduce some minimal delay between the calls so that `getStatusForTimestamp:` is called with more or less different timestamps? Would that make any difference?

Reply:
Good note. As it turns out with certain delay without changing current location `SBNavigator` starts behaving incorrectly. I'll submit test which covers that. I've also created ticket on nav. native side [here](https://github.com/mapbox/mapbox-navigation-native/issues/1445).

Reply:
> Good note. As it turns out with certain delay without changing current location `SBNavigator` starts behaving incorrectly. I'll submit test which covers that. I've also created ticket on nav. native side.